File: FoodDelivery.Web/src/app/models/restaurantOrder/RestaurantOrder.ts
import { OrderStatus } from './../enums/statuses/OrderStatus';
import { RestaurantOrderItem } from './RestaurantOrderItem';
import { Guid } from 'guid-typescript';

export class RestaurantOrder
{
    public id: Guid;
    public orderId: Guid;
    public status: OrderStatus;
    public restaurantName: string;
    public restaurantOrderItems: RestaurantOrderItem[];
    public totalSum: number;
}

File: FoodDelivery.DAL/EF/Entities/RestaurantOrder.cs
using System;
using System.Collections.Generic;

namespace FoodDelivery.DAL.EF.Entities
{
	public class RestaurantOrder
	{
		public Guid Id { get; set; }
		public Guid RestaurantId { get; set; }
		public Guid OrderId { get; set; }
		public int Status { get; set; }

		public virtual Restaurant Restaurant { get; set; }
		public virtual Order Order { get; set; }
		public virtual List<RestaurantOrderItem> RestaurantOrderItems { get; set; }
	}
}
